sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -577,45 +607,14 @@ object ScalaJSPluginInternal {
             "var __ScalaJSEnv = (typeof __ScalaJSEnv === \"object\" && __ScalaJSEnv) ? __ScalaJSEnv : {};\n" +
             "__ScalaJSEnv.javaSystemProperties = {" + formattedProps.mkString(", ") + "};\n"
           }
-          Seq(ResolvedJSDependency.minimal(
-              new MemVirtualJSFile("setJavaSystemProperties.js").withContent(code)))
+          Seq(new MemVirtualJSFile("setJavaSystemProperties.js").withContent(code))
         }
       },
 
-      loadedJSEnv := {
-        val log = streams.value.log
-        val env = resolvedJSEnv.value
-        val deps =
-          resolvedJSDependencies.value.data ++ scalaJSConfigurationLibs.value
+      // Crucially, add the Scala.js linked file to the JS files
+      scalaJSJSFiles += scalaJSLinkedFile.value,
```

Comment:
I have actually considered that for a good while, but it comes with its own set of issues.

* It makes it tricky to customize that list. You would want to customize it unscoped most of the time. But here we would have to use a stage-dispatch in the unscoped task, to be used by running environments, which means you can't customize it unscoped. You have to repeat the customization in the two tasks. 
* It creates a coupling from the jsEnv set of keys to the notion of stage task. That will force us to keep those things tightly coupled forever, but I'd like to investigate whether we can make JS envs less dependent on the Scala.js linker. Ideally they wouldn't care about Scala.js tools and anything related.